azure-iot-device/setup.py

```diff
@@ -27,13 +28,27 @@
 with open("README.md", "r") as fh:
     _long_description = fh.read()
 
-constant = {}
-with open("azure/iot/device/constant.py") as fh:
-    exec(fh.read(), constant)
+
+filename = "azure/iot/device/constant.py"
+version = None
+
+with open(filename, "r") as fh:
```

Comment:
You're reading constant.py twice, once to make sure version is included and again to extract version.  Why not just check to make sure version is not None after you extract it using re.search?

Reply:
+1

---
In reply to: [391009563](https://github.com/Azure/azure-iot-sdk-python/pull/482#discussion_r391009563) [](ancestors = 391009563)

Reply:
just wanted to fail fast in case not there...i originally had it together line by line...

---
In reply to: [391157717](https://github.com/Azure/azure-iot-sdk-python/pull/482#discussion_r391157717) [](ancestors = 391157717,391009563)